test/sol5/kyberNetworkProxy.js

```diff
@@ -578,6 +602,392 @@ contract('KyberNetworkProxy', function(accounts) {
+            generousNetwork2 =await GenerousNetwork2.new(admin);
+            await nwHelper.setupNetwork(generousNetwork2, networkProxy.address, KNC.address, DAO.address, tokens, accounts, admin, operator);
+            result = await nwHelper.setupReserves(generousNetwork2, tokens, 1, 1, 0, 0, accounts, admin, operator);
+            await nwHelper.addReservesToNetwork(generousNetwork2, result.reserveInstances, tokens, operator);
+        });
+
+        it("trade revert if src address is not eth and msg value is not zero", async () => {
+            let srcToken = tokens[1];
+            let srcQty = new BN(3).mul(new BN(10).pow(new BN(tokenDecimals[1])));
+
+            //get rate
+            let rate = await networkProxy.getExpectedRate(srcToken.address, tokens[2].address, srcQty);
+            dstQty = Helper.calcDstQty(srcQty, tokenDecimals[1], tokenDecimals[2], rate.expectedRate);
+            await srcToken.transfer(taker, srcQty);
+            await srcToken.approve(networkProxy.address, srcQty, { from: taker });
+
+            //see trade reverts 
+            await expectRevert.unspecified(networkProxy.trade(
+                srcToken.address,
+                srcQty,
+                tokens[2].address,
+                taker,
+                maxDestAmt,
+                rate.expectedRate,
+                zeroAddress,
+                { from: taker, value: new BN(2) }
+            ));
+            // see trade is ok if msg.value == 0
+            await networkProxy.trade.call(
+                srcToken.address,
+                srcQty,
+                tokens[2].address,
+                taker,
+                maxDestAmt,
+                rate.expectedRate,
+                zeroAddress,
+                { from: taker }
+            );
+        });
+
+        it("trade revert if actual dest amount is different from reported", async () => {
+            await networkProxy.setKyberNetwork(generousNetwork.address, { from: admin });
+            let srcToken = tokens[1];
+            let dstToken = tokens[2]
+            let amountTwei = 1515;
+
+            tokenAmountForTrades = new BN(10000).mul(new BN(10).pow(tokenDecimals[1]));
+            await srcToken.transfer(taker, amountTwei);
+            await dstToken.transfer(generousNetwork.address, tokenAmountForTrades);
+            //get rate
+            let rate = await networkProxy.getExpectedRate(srcToken.address, dstToken.address, amountTwei);
+            await srcToken.approve(networkProxy.address, amountTwei, { from: taker });
+
+            //see trade reverts
+            await expectRevert(
+                networkProxy.trade(
+                    srcToken.address,
+                    amountTwei,
+                    dstToken.address,
+                    taker,
+                    maxDestAmt,
+                    rate.expectedRate,
+                    zeroAddress,
+                    { from: taker }
+                ), "wrong ret amount"
```

Comment:
please change revert message in network to:
"Kyber network returned wrong amount."

and change also here.

Reply:
updated in https://github.com/KyberNetwork/smart-contracts/pull/726/commits/e4d067f71b9400d43a678940a586bbb43991f02a